src/libstore/gc.cc

```diff
@@ -623,8 +633,8 @@ void LocalStore::collectGarbage(const GCOptions & options, GCResults & results)
            by another process. We need to be sure that we can acquire an
            exclusive lock before deleting them. */
         if (baseName.find("tmp-", 0) == 0) {
-            AutoCloseFD tmpDirFd = open(realPath.c_str(), O_RDONLY | O_DIRECTORY);
-            if (tmpDirFd.get() == -1 || !lockFile(tmpDirFd.get(), ltWrite, false)) {
+            AutoCloseFD tmpDirFd = openDirectory(realPath);
+            if (tmpDirFd.get() == INVALID_DESCRIPTOR || !lockFile(tmpDirFd.get(), ltWrite, false)) {
```

Comment:
Maybe it would make sense to have a `(bool)` conversion or `operator !` so we could write
```suggestion
            if (!tmpDirFd|| !lockFile(tmpDirFd.get(), ltWrite, false)) {
```
or otherwise an `isValid()` method.

Reply:
oh I think we do have that lol